source/userguide/entity_resolver.rst

```diff
@@ -171,51 +171,230 @@ Create the synonyms and make them available to your MindMeld app, going entity t
+
+.. code-block:: python
+
+   nlp.domains['ordering'].intents['build_order'].build()
+   er = nlp.domains['ordering'].intents['build_order'].extract_entities()['dish'].entity_resolver
+   er
+
+.. code-block:: console
+
+   <ElasticsearchEntityResolver ready: False, dirty: False>
+
+Alternatively, build the NLP pipeline using :meth:`NaturalLanguageProcessor.build()`. Among its other tasks, this initializes as well as fits the resolver. When you run :meth:`NaturalLanguageProcessor.process()`, the NLP pipeline includes the resolved entities in its results.
+
+.. code-block:: python
+
+   nlp.domains['ordering'].intents['build_order'].build()
+   er = nlp.domains['ordering'].intents['build_order'].entities['dish'].entity_resolver
+   er
+
+.. code-block:: console
+
+   <ElasticsearchEntityResolver ready: False, dirty: False>
+
+Access through `EntityResolverFactory` module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One can also use an entity resolver independent of the :mod:`nlp` pipeline. This could be useful to train/evaluate performances of different resolvers on your dataset(s). To do so, import `EntityResolverFactory` module as follows:
+
+.. code-block:: python
+
+   from mindmeld.components import EntityResolverFactory
+   er = EntityResolverFactory(app_path='food_ordering', entity_type='dish')
+   er
+
+.. code-block:: console
+
+   <ElasticsearchEntityResolver ready: False, dirty: False>
+
+If no resolver configs are specified, the module loads a resolver based on default configs, which involves `Elasticsearch <https://www.elastic.co/products/elasticsearch>`_ based entity resolution. Instead, one can specify the type of resolver they would like to use in the arguments of `EntityResolverFactory` as follows:
+
+.. code-block:: python
+
+    ENTITY_RESOLVER_CONFIG = {
+        'model_type': 'resolver',
+        "model_settings": {
+            "resolver_type": "sbert_cosine_similarity",
+        }
+    }
+    er = EntityResolverFactory.create_resolver(app_path='food_ordering', entity_type='dish', er_config=ENTITY_RESOLVER_CONFIG)
+    er
+
+.. code-block:: console
+
+   <SentenceBertCosSimEntityResolver ready: False, dirty: False>
 
 Train the entity resolver
 -------------------------
 
-The Entity Resolver uses the `Elasticsearch <https://www.elastic.co/products/elasticsearch>`_ full-text search and analytics engine for information retrieval. This is required by the advanced text similarity model MindMeld uses by default. Once Elasticsearch is up and running, no configuration needed is needed within MindMeld. To learn how to set up Elasticsearch, see the :doc:`Getting Started guide <getting_started>`.
+Use the :meth:`.fit()` method to train a entity resolution model. Depending on the size of the training data, this can take anywhere from a few seconds to several minutes. With logging level set to ``INFO`` or below, you should see the build progress in the console.
```

Comment:
I think it is probably better to have different sections - Entity Resolution with ElasticSearch and Entity Resolution without ES. That way we can give clear descriptions of what happens during fit(), clean fit, etc. Right now, that's a little hard to follow.